### PR TITLE
Implemented functionality for scan button PO details page ( #1xx6hbj )

### PR DESCRIPTION
**Pull request by meet-aniket**

### CODE REVIEW

File: src/store/modules/purchaseOrder/mutation-types.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export const SN_PURCHASE_ORDER = 'purchase_order'
 export const PURCHASE_ORDER_UPDATED = SN_PURCHASE_ORDER + '/UPDATED'
-export const PURCHASE_ORDER_DETAIL_UPDATED = SN_PURCHASE_ORDER+ '/DETAIL_UPDATED'
\ No newline at end of file
+export const PURCHASE_ORDER_DETAIL_UPDATED = SN_PURCHASE_ORDER+ '/DETAIL_UPDATED'
+export const UPDATE_PO_PRODUCT_COUNT = SN_PURCHASE_ORDER + '/COUNT_UPDATED'
```
Comment (ymaheshwari1) [inline src/store/modules/purchaseOrder/mutation-types.ts:4]: Improve mutation type name to have consistency among names.

File: src/views/PurchaseOrderDetail.vue [hunk 2]
```diff
@@ -18,20 +18,25 @@
...
-        </ion-item>
-        <ion-item>
-          <ion-label>{{$t("Scan Items")}}</ion-label>
+          <ion-label>{{$t("Purchase Order")}}: {{ orderDetail[0]?.externalOrderId }}</ion-label>
         </ion-item>
+        <div class="po-scanner">
+          <ion-item class="action-text">
+            {{$t("Scan Items")}}
+            <ion-label>
+              <ion-input :placeholder="$t('Scan barcodes to receive them')"></ion-input>
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:27]: Use self-closing tag

File: src/views/PurchaseOrderDetail.vue [hunk 3]
```diff
@@ -139,12 +145,27 @@ export default defineComponent({
...
+      this.store.dispatch('purchaseOrder/updatePoProductCount', payload)
+    },
+    async scanCode(){
+      const modal = await modalController
+        .create({
+          component: Scanner,
+        });
+        modal.onDidDismiss()
+          .then((result) => {
+            this.updateProductCount(result.role);
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:159]: Why we are calling a method here and defining a new method? We can directly dispatch the action from here.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/purchaseOrder/mutation-types.ts (hunk 1 above)
Comment (meet-aniket) [reply src/store/modules/purchaseOrder/mutation-types.ts:4]: Sir, I have changed the variable name as:- PURCHASE_ORDER_PRODUCT_COUNT_UPDATED

File: src/views/PurchaseOrderDetail.vue (hunk 2 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:27]: Changed Sir,

**Pull request merged**